cli/ping.go

```diff
@@ -77,11 +135,63 @@ func (r *RootCmd) ping() *serpent.Command {
 			defer conn.Close()
 
 			derpMap := conn.DERPMap()
-			_ = derpMap
 
+			diagCtx, diagCancel := context.WithTimeout(inv.Context(), 30*time.Second)
+			defer diagCancel()
+			diags := conn.GetPeerDiagnostics()
```

Comment:
Since this is a network troubleshooting command we have to be especially concerned about responsiveness during a slow connection. 30s is too long to wait for the first ping here. I think we either reduce the timeout, or add a background job that emits something like "still collecting diagnostics... slow connection" every 3 seconds it pends.

Reply:
Yeah, there's like a good two second delay before the pings start logging - the spinner we use when uploading templates seems pretty good here, if you're not opposed to that:

https://github.com/user-attachments/assets/9dd0d7aa-4889-4148-b586-b2281f1adc75

Reply:
Ah I like that.